src/coreclr/jit/compiler.hpp

```diff
@@ -687,6 +733,11 @@ BasicBlockVisit BasicBlock::VisitAllSuccs(Compiler* comp, TFunc func, const bool
         case BBJ_SWITCH:
         {
             Compiler::SwitchUniqueSuccSet sd = comp->GetDescriptorForSwitch(this);
+            jitstd::sort(sd.nonDuplicates, (sd.nonDuplicates + sd.numDistinctSuccs),
+                         [](FlowEdge* const lhs, FlowEdge* const rhs) {
+                return (lhs->getLikelihood() * lhs->getDupCount()) < (rhs->getLikelihood() * rhs->getDupCount());
+            });
```

Comment:
It seems odd for the visitor function to be mutating the descriptor.

It would make more sense to me if this sorting step was done ahead of time as part of the block layout phase.

Reply:
> It would make more sense to me if this sorting step was done ahead of time as part of the block layout phase.

Are you thinking of something like a pass over the block list that sorts all the switch descriptors, before computing the DFS? If so, considering the diffs don't look all that significant, I'm tempted to just visit switch successors in their unsorted order to avoid cluttering the visitor APIs, or adding another loop to the layout algorithm that won't do anything most of the time.

Reply:
Something like that, or if that is undesirable, then some map created lazily in the context of the block layout phase (i.e. when we visit the switch block). I guess the latter is not much different to just allocating new memory to sort these successors into.
But at that point it makes more sense to me if we change `AllSuccessorEnumerator` slightly: in reality it's really just a vector with small vector optimization for `N=4`. If you phrase `fgRunDfs` in terms of a callback that returns this small vector of successors, then the memory where we can do the sorting without mutating existing data structures will already be available.

Reply:
> If so, considering the diffs don't look all that significant, I'm tempted to just visit switch successors in their unsorted order to avoid cluttering the visitor APIs

I would suggest this, unless you have some information that order of successors matters for switches.